great_expectations/data_context/config_validator/yaml_config_validator.py

```diff
@@ -117,9 +117,6 @@ def test_yaml_config(  # noqa: PLR0912, PLR0913
         class_name: Optional[str] = None,
         runtime_environment: Optional[dict] = None,
         pretty_print: bool = True,
-        return_mode: Literal[
-            "instantiated_class", "report_object"
-        ] = "instantiated_class",
```

Comment:
Removing `self_check` so removing `report_object` (meaning we no longer need this param)